### PR TITLE
[arser] Implement lexical_cast for boolean type

### PR DESCRIPTION
**Pull request by mhs4670go**
This commit implements lexical_cast for boolean type

ONE-DCO-1.0-Signed-off-by: seongwoo <mhs4670go@naver.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
Where is this used?

**Comment by mhs4670go** [comment]
@seanshpark Actually, there is an error when boolean type is passed from user as of now. For bug fix.

**Comment by ragmani** [comment]
@nnfw-bot test nncc-release

**Comment by seanshpark** [comment]
> For bug fix.

Then it would be nice if you could add a link or add some explanation about the error.

File: compiler/arser/include/arser/arser.h [hunk 1]
```diff
@@ -40,6 +40,13 @@ template <typename T> T lexical_cast(const std::string &str)
   ss >> data;
   return data;
 }
+template <> bool lexical_cast(const std::string &str)
```
Comment (seanshpark) [inline compiler/arser/include/arser/arser.h:43]: please add an empty space in between

**Comment by mhs4670go** [comment]
@seanshpark You're right.
FYI, when user "get"s the value that he registered, "get" method calls `lexical_cast`.
```cxx
template <typename T> T lexical_cast(const std::string &str)
{
  std::istringstream ss;
  ss.str(str);
  T data;
  ss >> data;
  return data;
}
```
But this implementation can't cover boolean type. This is why I've posted this PR:)

**Pull request merged**